compiler/noirc_driver/src/lib.rs

```diff
@@ -56,6 +56,12 @@ pub struct CompileOptions {
     #[arg(long, value_parser = parse_expression_width, default_value = "4")]
     pub expression_width: ExpressionWidth,
 
+    /// Generate ACIR without an expression width bound.
+    /// This should usually only be used by users looking to hand optimize
+    /// their circuits using the `as_witness` builtin.
```

Comment:
Can we add something along the lines of "Note that this regardless of the value of this flag, any expressions will be split in order to conform with `expression_width`?

Reply:
I changed the comment a bit as now this flag is `bounded_codegen` as the default is to have unbounded codegen.